src/drive/share/index.ts

```diff
@@ -0,0 +1,89 @@
+import axios, { AxiosStatic } from 'axios';
+import { Token } from '../../auth';
+import { headersWithTokenAndMnemonic } from '../../shared/headers';
+import { GenerateShareLinkPayload, GetShareInfoResponse, IShare } from './types';
+
+export * as ShareTypes from './types';
+
+export class Share {
+  private readonly axios: AxiosStatic;
+  private readonly apiUrl: string;
+  private readonly clientName: string;
+  private readonly clientVersion: string;
+  private readonly token: Token;
+  private readonly mnemonic: string;
+
+  public static client(
+    apiUrl: string, clientName: string, clientVersion: string, token: Token, mnemonic: string
+  ) {
+    return new Share(axios, apiUrl, clientName, clientVersion, token, mnemonic);
+  }
+
+  constructor(
+    axios: AxiosStatic, apiUrl: string, clientName: string, clientVersion: string, token: Token, mnemonic: string
+  ) {
+    this.axios = axios;
+    this.apiUrl = apiUrl;
+    this.clientName = clientName;
+    this.clientVersion = clientVersion;
+    this.token = token;
+    this.mnemonic = mnemonic;
+  }
+
+  /**
+   * Generates a new link to share a file
+   * @param payload
+   */
+  public generateShareLink(payload: GenerateShareLinkPayload): Promise<{
+    token: string
+  }> {
+    return this.axios
+      .post(`${this.apiUrl}/api/storage/share/file/${payload.fileId}`, {
+        isFolder: payload.isFolder,
+        views: payload.views,
+        encryptionKey: payload.encryptionKey,
+        fileToken: payload.fileToken,
+        bucket: payload.bucket,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+  }
+
+  /**
+   * Fetches info about a specific token
```

Comment:
"Fetches info about a specific share" is more correct. The token is only used as an indexed field to retrieve the required info (the share) to recover the decryption key and download a file